src/app/screens/details_actions.rs

```diff
@@ -338,34 +393,48 @@ impl DetailsActions {
         config.git_am_options().split_whitespace().for_each(|opt| {
             git_am_out.arg(opt);
         });
-        let git_am_out = git_am_out.output().unwrap();
+
+        let git_am_out = git_am_out
+            .output()
+            .map_err(|e| format!("failed to execute git-am: {}", e))?;
+
         if !git_am_out.status.success() {
             let _ = Command::new("git")
                 .arg("-C")
                 .arg(kernel_tree.path())
                 .arg("am")
                 .arg("--abort")
                 .output()
-                .unwrap();
+                .map_err(|e| format!("failed to abort git-am: {}", e));
+
+            return Err(String::from_utf8_lossy(&git_am_out.stderr).to_string());
         }
 
-        // 9. Return back to original branch
-        let _ = Command::new("git")
-            .arg("-C")
-            .arg(kernel_tree.path())
-            .arg("switch")
-            .arg(&original_branch)
-            .output()
-            .unwrap();
+        Ok(())
+    }
 
-        if git_am_out.status.success() {
-            Ok(format!(" Patchset '{}' applied successfully!\n\n - Kernel Tree: '{}'\n\n - Base Branch: '{}'\n\n - Applied branch: '{}'", self.representative_patch.title(), kernel_tree.path(), kernel_tree.branch(), &target_branch_name, ))
-        } else {
-            Err(format!(
-                "`git am` failed\n{}{}",
-                &original_branch,
-                String::from_utf8_lossy(&git_am_out.stderr)
-            ))
+    /// Try to apply the patchset to a target kernel tree and returns a `String`
+    /// informing if the apply succeeded or failed and why.
+    ///
+    ///
+    /// Returns a `String` containing either the success or error message.
```

Comment:
Adapt the return value

```suggestion
    /// Returns a `Result<String, String>` containing either the success or the error message.
```